python/packages/autogen-core/src/autogen_core/tools/_function_tool.py

```diff
@@ -87,6 +88,12 @@ def __init__(
     ) -> None:
         self._func = func
         self._global_imports = global_imports
+
+        # Validate type hints exist
+        hints = typing.get_type_hints(func)
+        if not hints:
+            raise TypeError(f"Function '{func.__name__}' has no type hints. All parameters must be annotated.")
```

Comment:
The issue you referenced, the function argument has type hint. Can this catch the same error?